core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2204,11 +2209,22 @@ private void processTxFromBestChain(Transaction tx, boolean forceAddToPool) thro
             addWalletTransaction(Pool.SPENT, tx);
         }
 
-        // Kill txns in conflict with this tx
-        Set<Transaction> doubleSpendTxns = findDoubleSpendsAgainst(tx, pending);
-        if (!doubleSpendTxns.isEmpty()) {
-            // no need to addTransactionsDependingOn(doubleSpendTxns) because killTxns() already kills dependencies;
-            killTxns(doubleSpendTxns, tx);
+        if(!findDoubleSpendsAgainst(tx, transactions).isEmpty()) {
```

Comment:
I realised now that this check might not be necessary anymore, since the transaction at this point is either a double spend or has outputs to/from wallet, in which case it will be ignored below.